File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/model/MissingArgumentError.java
package org.openapitools.vertxweb.server.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.openapitools.vertxweb.server.model.CodedErrorBase;
import org.openapitools.vertxweb.server.model.MissingArgumentErrorAllOf;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class MissingArgumentError   {
  
  private Object result = null;
  private Object msg = null;
  private Object code = null;
  private String varName;

  public MissingArgumentError () {

  }

  public MissingArgumentError (Object result, Object msg, Object code, String varName) {
    this.result = result;
    this.msg = msg;
    this.code = code;
    this.varName = varName;
  }

    
  @JsonProperty("result")
  public Object getResult() {
    return result;
  }
  public void setResult(Object result) {
    this.result = result;
  }

    
  @JsonProperty("msg")
  public Object getMsg() {
    return msg;
  }
  public void setMsg(Object msg) {
    this.msg = msg;
  }

    
  @JsonProperty("code")
  public Object getCode() {
    return code;
  }
  public void setCode(Object code) {
    this.code = code;
  }

    
  @JsonProperty("var_name")
  public String getVarName() {
    return varName;
  }
  public void setVarName(String varName) {
    this.varName = varName;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MissingArgumentError missingArgumentError = (MissingArgumentError) o;
    return Objects.equals(result, missingArgumentError.result) &&
        Objects.equals(msg, missingArgumentError.msg) &&
        Objects.equals(code, missingArgumentError.code) &&
        Objects.equals(varName, missingArgumentError.varName);
  }

  @Override
  public int hashCode() {
    return Objects.hash(result, msg, code, varName);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MissingArgumentError {\n");
    
    sb.append("    result: ").append(toIndentedString(result)).append("\n");
    sb.append("    msg: ").append(toIndentedString(msg)).append("\n");
    sb.append("    code: ").append(toIndentedString(code)).append("\n");
    sb.append("    varName: ").append(toIndentedString(varName)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: cpp-tiny/lib/Models/MissingArgumentError.cpp


#include "MissingArgumentError.h"

using namespace Tiny;

MissingArgumentError::MissingArgumentError()
{
	result = null;
	msg = null;
	code = null;
	var_name = std::string();
}

MissingArgumentError::MissingArgumentError(std::string jsonString)
{
	this->fromJson(jsonString);
}

MissingArgumentError::~MissingArgumentError()
{

}

void
MissingArgumentError::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *resultKey = "result";
    
    if(object.has_key(resultKey)) 
    {
        bourne::json value = object[resultKey];


        

        AnyType* obj = &result;
		obj->fromJson(value.dump());

    }

    const char *msgKey = "msg";
    
    if(object.has_key(msgKey)) 
    {
        bourne::json value = object[msgKey];


        

        AnyType* obj = &msg;
		obj->fromJson(value.dump());

    }

    const char *codeKey = "code";
    
    if(object.has_key(codeKey)) 
    {
        bourne::json value = object[codeKey];


        

        AnyType* obj = &code;
		obj->fromJson(value.dump());

    }

    const char *var_nameKey = "var_name";
    
    if(object.has_key(var_nameKey)) 
    {
        bourne::json value = object[var_nameKey];


        
        jsonToValue(&var_name, value, "std::string");


    }


}

bourne::json
MissingArgumentError::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["result"] = getResult().toJson();

    




	object["msg"] = getMsg().toJson();

    




	object["code"] = getCode().toJson();

    



    object["var_name"] = getVarName();



    return object;

}

AnyType
MissingArgumentError::getResult()
{
	return result;
}

void
MissingArgumentError::setResult(AnyType  result)
{
	this->result = result;
}

AnyType
MissingArgumentError::getMsg()
{
	return msg;
}

void
MissingArgumentError::setMsg(AnyType  msg)
{
	this->msg = msg;
}

AnyType
MissingArgumentError::getCode()
{
	return code;
}

void
MissingArgumentError::setCode(AnyType  code)
{
	this->code = code;
}

std::string
MissingArgumentError::getVarName()
{
	return var_name;
}

void
MissingArgumentError::setVarName(std::string  var_name)
{
	this->var_name = var_name;
}



